src/lib.rs

```diff
@@ -2438,6 +2438,12 @@ impl Connection {
 
         self.sent_count += 1;
 
+        if self.dgram_queue.writable().pending_bytes() >
+            self.recovery.cwnd_available()
+        {
+            self.recovery.update_app_limited(false);
+        }
+
```

Comment:
↑ this is the crux of the issue. Without this the unit test doesn't work, and this here is a little inelegant. Might come a little better refactoring it into a separate function, but that's just adding a little sugar to make it less sour.